src/main/clojure/clara/rules/compiler.clj

```diff
@@ -235,15 +235,21 @@
   "Compiles a sequence of constraints into a structure that can be evaluated.
 
    Callers may also pass a collection of equality-only-variables, which instructs
-   this function to only do an equality check on them rather than create a unification binding."
+   this function to only do an equality check on them rather than create a unification binding.
+
+   The returned expression evaluates to a two item vector, [bindings error], which contains
+   the evaluated bindings in the first argument and a map with information on any exception
+   that might have been thrown by evaluating a constraint as the second vector entry."
   ([exp-seq]
-   (compile-constraints exp-seq #{}))
+   (compile-constraints exp-seq #{} 0))
   ([exp-seq equality-only-variables]
+   (compile-constraints exp-seq equality-only-variables 0))
+  ([exp-seq equality-only-variables constraint-idx]
```

Comment:
Agreed, the original implementation published constraint-idx and the node reconciled it. There is no need for constraint-idx now.